### ISSUE TITLE
ResultSet findColumn on empty result

### ISSUE THREAD

**Issue by gwenn** [xerial/sqlite-jdbc#744]
```patch
diff --git a/src/test/java/org/sqlite/ResultSetTest.java b/src/test/java/org/sqlite/ResultSetTest.java
index aa8d2d6..f095576 100644
--- a/src/test/java/org/sqlite/ResultSetTest.java
+++ b/src/test/java/org/sqlite/ResultSetTest.java
@@ -42,6 +42,13 @@ public class ResultSetTest {
         assertEquals(1, resultSet.findColumn("id"));
     }

+    @Test
+    public void testFindColumnOnEmptyResultSet() throws SQLException {
+        ResultSet resultSet = stat.executeQuery("select * from test where id = 0");
+        assertFalse(resultSet.next());
+        assertEquals(1, resultSet.findColumn("id"));
+    }
+
     @Test
     public void testTableColumnLowerNowFindUpperCaseColumn() throws SQLException {
         ResultSet resultSet = stat.executeQuery("select * from test");
```
```sh
sqlite-jdbc % mvn test
...
[ERROR] Tests run: 16, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.066 s <<< FAILURE! - in org.sqlite.ResultSetTest
[ERROR] testFindColumnOnEmptyResultSet  Time elapsed: 0.011 s  <<< ERROR!
java.sql.SQLException: ResultSet closed
	at org.sqlite.ResultSetTest.testFindColumnOnEmptyResultSet(ResultSetTest.java:49)
```

**Comment by gotson** [comment]
Can you provide details of what this is about?

**Comment by gwenn** [comment]
I am not sure what details you need but...
One example:
```java
try (ResultSet rs = null/*...*/) {
  int nameIdx = rs.findColumn("name"); // lookup once and don't fail if rs is empty
  // and then iterate
  while (rs.next()) {
    String name = rs.getString(nameIdx);
    //...
  }
}
```
Or take a look at [h2 testFindColumn](https://github.com/h2database/h2database/blob/master/h2/src/test/org/h2/test/jdbc/TestResultSet.java#L494-L530).

**Comment by gotson** [comment]
> I am not sure what details you need

Your issue doesn't mention anything, just throw some code.

Can you clarify if this is a bug, a feature request ?

**Comment by gwenn** [comment]
For me, this is a bug.
ResultSet is obviously not closed, just empty.
But still, `sqlite-jdbc` throws `java.sql.SQLException: ResultSet closed`.

**Comment by gotson** [comment]
do you have any experience with other JSBC drivers on that front ?

**Comment by gwenn** [comment]
Yes, https://github.com/gwenn/sqlite-jna.

**Comment by gotson** [comment]
that's a bit biased, isn't it ? :)

**Comment by pyckle** [comment]
I believe @gwenn  is correct that this is a bug. The JDBC specification indicates that findColumn throws an exception when the ResultSet is closed. However, this is only defined to happen when the parent statement is closed, or an explicit call to close() is made. See the javadoc here: https://docs.oracle.com/javase/7/docs/api/java/sql/ResultSet.html

**Comment by gotson** [comment]
i've created #764 to fix that. I think the problem comes from [here](https://github.com/xerial/sqlite-jdbc/pull/764/files#diff-4eb4ac21994d625e5023c2751448f8c2f509cbfc43f99f143f19c3cfa074d546R120), where a result set would be returned as closed because it has empty results.

Even though it's empty, i think it should still be open.

@pyckle @gwenn would appreciate your thoughts on this.

**Comment by gwenn** [comment]
Thanks for the fix.
I don't know where the `open` flag is used.
But definitely, an empty ResultSet is not closed (until its `close()` method is called).
I guess the following [comment](https://github.com/xerial/sqlite-jdbc/blob/master/src/main/java/org/sqlite/core/CoreResultSet.java#L29) should be updated:
```java
    public boolean open = false; // true means have results and can iterate them
```
And, not related but it seems that the following [comment](https://github.com/xerial/sqlite-jdbc/blob/master/src/main/java/org/sqlite/core/CoreResultSet.java#L28) is wrong:
```java
    public boolean emptyResultSet = false; // false means we don't have results
```

**Comment by gotson** [comment]
> I don't know where the `open` flag is used.

The tests showed some usage in some functions, and I fixed that. Could be more though.

---

i have also updated the comments.

**Comment by pyckle** [comment]
Looks good to me. Thank you for fixing my incorrect comment (:

**Comment by gotson** [comment]
Closed by #764
